search/search_repository_whitebox_test.go

```diff
@@ -148,36 +157,76 @@ func (s *searchRepositoryWhiteboxTest) TestSearch() {
 		verify(t, searchQuery, searchResults, 0)
 	})
 
-	s.T().Run("Search by number", func(t *testing.T) {
+	s.T().Run("Search by number - single match", func(t *testing.T) {
+		// given
+		fxt := tf.NewTestFixture(t, s.DB, tf.WorkItems(10))
+		queryNumber := fxt.WorkItems[2].Number
+		// when looking for `number:3`
+		searchQuery := fmt.Sprintf("number:%d", queryNumber)
+		spaceID := fxt.Spaces[0].ID.String()
+		searchResults, _, err := s.sr.SearchFullText(context.Background(), searchQuery, &start, &limit, &spaceID)
+		// then there should be a single match
+		require.Nil(t, err)
+		require.Len(t, searchResults, 1)
+		assert.Equal(t, queryNumber, searchResults[0].Number)
+	})
+
+	s.T().Run("Search by number - multiple matches", func(t *testing.T) {
```

Comment:
What do you think of nesting those tests to get a better context and not repeat yourself?

```go
s.T().Run("by number", func(t*testing.T)) {
    t.Run("single match", func(t*testing.T)) {
    })
    t.Run("multiple match", func(t*testing.T)) {
    })
})
s.T().Run("by URL", func(t*testing.T)) {
    t.Run("single match", func(t*testing.T)) {
    })
    t.Run("multiple match", func(t*testing.T)) {
    })
})
```